util/src/main/java/io/kubernetes/client/PortForward.java

```diff
@@ -0,0 +1,180 @@
+package io.kubernetes.client;
+
+import io.kubernetes.client.Configuration;
+import io.kubernetes.client.models.V1Pod;
+import io.kubernetes.client.util.WebSockets;
+import io.kubernetes.client.util.WebSocketStreamHandler;
+
+import java.io.InputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+/**
+ * Utility class for setting up port-forwarding connections.
+ * Uses the WebSockets API, not the SPDY API (which the Go client uses)
+ * 
+ * The protocol is undocumented as far as I can tell, but the PR that added
+ * it is here:
+ * https://github.com/kubernetes/kubernetes/pull/33684
+ * 
+ * And the protocol is:
+ * 
+ * ws://server/api/v1/namespaces/<namespace>/pods/<pod>/portforward?ports=80&ports=8080
+ * 
+ * I/O for first port (80) is on Channel 0
+ * Err for first port (80) is on Channel 1
+ * I/O for second port (8080) is on Channel 2
+ * Err for second port (8080) is on Channel 2
+ * <and so on for remaining ports>
+ * 
+ * The first two bytes of each output stream is the port that is being forwarded
+ * in little-endian format.
+ */
+public class PortForward {
+    private ApiClient apiClient;
+
+    /**
+     * Simple PortForward API constructor, uses default configuration
+     */
+    public PortForward() {
+        this(Configuration.getDefaultApiClient());
+    }
+
+    /**
+     * PortForward API Constructor
+     * @param apiClient The api client to use.
+     */
+    public PortForward(ApiClient apiClient) {
+        this.apiClient = apiClient;
+    }
+
+    /**
+     * Get the API client for these PortForward operations.
+     * @returns The API client that will be used.
+     */
+    public ApiClient getApiClient() {
+        return apiClient;
+    }
+
+    /**
+     * Set the API client for subsequent PortForward operations.
+     * @param apiClient The new API client to use.
+     */
+    public void setApiClient(ApiClient apiClient) {
+        this.apiClient = apiClient;
+    }
+
+    private String makePath(String namespace, String name) {
+          return "/api/v1/namespaces/" +
+            namespace +
+            "/pods/" +
+            name +
+            "/portforward";
+    }
+
+    /**
+     * PortForward to a container
+     * 
+     * @param pod The pod where the port forward is run.
+     * @param ports The ports to forward
+     */
+    public PortForwardResult forward(V1Pod pod, List<Integer> ports) throws ApiException, IOException {
+        return forward(pod.getMetadata().getNamespace(), pod.getMetadata().getNamespace(), ports);
+    }
+
+    /**
+     * PortForward to a container.
+     * 
+     * @param namespace The namespace of the Pod
+     * @param name The name of the Pod
+     * @param ports The ports to forward
```

Comment:
Please add `@return`, I don't think our linter is catching it but the release will fail w/o it.

Reply:
fixed.